ddl/ddl_worker.go

```diff
@@ -451,6 +465,19 @@ func newMetaWithQueueTp(txn kv.Transaction, tp string) *meta.Meta {
 	return meta.NewMeta(txn)
 }
 
+func (w *worker) setDDLLabelForTopSQL(job *model.Job) {
+	if !variable.TopSQLEnabled() || job == nil {
+		return
+	}
+
+	if job.Query != w.cacheSQL {
+		w.cacheNormalizedSQL, w.cacheDigest = parser.NormalizeDigest(job.Query)
+		w.cacheSQL = job.Query
```

Comment:
exist a potential of attching it twice when owner changed. it that matters

Reply:
It's ok, `AttachSQLInfo` function is idempotent.